### PR TITLE
Some smaller refactorings

### PR DESCRIPTION
**Pull request by timhabermaas**
This is just some stuff I found while getting to know the code base. IMO none of the changes should be controversial and the PR is mergeable as is, unless Travis disagrees with me.

### CODE REVIEW

**Comment by TeatroIO** [comment]
I've prepared a stage to preview changes. <a href='http://timhabermaas-random-cleanups.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Open stage</a> or <a href='http://timhabermaas-random-cleanups.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/teatro.log' target='_blank'>view logs</a>.

**Comment by timhabermaas** [comment]
Thinking about it: d0426dcf1784c1e11c9bb262de11582a1060e6e8 probably conflicts with the Rails 4 upgrade. I can remove that commit if it causes conflicts.

File: app/services/create_work_package_service.rb [hunk 1]
```diff
@@ -43,7 +43,7 @@ def create
       author: user,
       type: project.types.where(is_default: true).first || project.types.first
     }
-    self.work_package = project.add_work_package(hash)
```
Comment (NobodysNightmare) [inline app/services/create_work_package_service.rb:46]: good catch! I thought, that we stopped remembering the own work package, long ago...

File: app/controllers/work_packages/auto_completes_controller.rb [hunk 2]
```diff
@@ -54,15 +54,15 @@ def index
                       .limit(10)
 
     respond_to do |format|
-      format.html do render layout: false end
-      format.any(:xml, :json) { render request.format.to_sym => wp_hashes_with_string }
+      format.html { render layout: false }
```
Comment (NobodysNightmare) [inline app/controllers/work_packages/auto_completes_controller.rb]: This seems to be the only change where I disagree:

We chose to distinguish between `do ... end` and `{ ... }` based on semantics. Best match on the internet might be [this one](https://github.com/bbatsov/ruby-style-guide/issues/162), which has some sources itself.

Basically: When the content of the block is about the side-effect, use `do ... end`, when it is about the return-value use `{ ... }`.

edit: I believe the heavy normalization work, especially regarding `format.x do render y end` was done in the Rails 4 branch...

**Comment by NobodysNightmare** [comment]
> d0426dc probably conflicts with the Rails 4 upgrade. I can remove that commit if it causes conflicts.

Nice to see, that you are thinking about that. But we can't stop introducing `find_by_foo` statements and your code is definitely an improvement (on the current codebase). I don't think that there will be hard git conflicts, though. Just another `find_by_foo` that needs to be replaced again :/

File: app/controllers/work_packages/auto_completes_controller.rb (hunk 2 above)
Comment (timhabermaas) [inline app/controllers/work_packages/auto_completes_controller.rb]: Ah, I see. I actually like that style guide, but no one ever used it. So, :+1:.

> edit: I believe the heavy normalization work, especially regarding format.x do render y end was done in the Rails 4 branch...

Then I better get rid of my commit. No need to risk conflicts for such a minor change.

**Comment by timhabermaas** [comment]
@NobodysNightmare I removed the block style change and force pushed the branch. Feel free to merge it.

**Comment by NobodysNightmare** [comment]
I feel that I merged this PR prematurely

image

Investigating...

edit: Okay, it has apparently been an error when merging this PR with the rails4 branch...

**Comment by basiszwo** [comment]
so what's the state here? still an error or not?

**Comment by NobodysNightmare** [comment]
Oh, nono... my edit was probably not clear enough.

It was an error with the merge immediately following the merge of this PR. And the error was fixed immediately. So nothing to worry about on this PR :)

**Pull request merged**